include/fuzzuf/executor/traits/afl_capable.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+/**
+ * @file afl_capable.hpp
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
+ */
+
+#ifndef FUZZUF_INCLUDE_EXECUTOR_TRAITS_AFL_CAPABLE_HPP
+#define FUZZUF_INCLUDE_EXECUTOR_TRAITS_AFL_CAPABLE_HPP
+
+#include "fuzzuf/executor/native_linux_executor.hpp"
+#include "fuzzuf/feedback/exit_status_feedback.hpp"
+#include "fuzzuf/feedback/inplace_memory_feedback.hpp"
+#include "fuzzuf/utils/common.hpp"
+
+/**
+ * @class DynAFLCapable
+ * @brief Represents minimal requirements for an AFL-capable executor.
+ * 
+ * @details The executor for AFL must have the methods declared in this class.
+ * This class is to perform type erasure for the executor class to abstract
+ * the executor on the algortihms. We found that the boost's type_erasure does
+ * not meet our needs, so we implemented our own type erasure.
+ */
+class DynAFLCapable
```

Comment:
I want to think of another name if possible because this name doesn't express that this is a container for "executors"...

Reply:
Indeed. Then I think we can leave this as it is.